From 1278aea063bbaa0c03f1ff69990de4a4ef69b4c1 Mon Sep 17 00:00:00 2001
From: Brian Bockelman <bbockelman@morgridge.org>
Date: Wed, 15 Jan 2025 08:03:48 -0600
Subject: [PATCH 2/8] XrdHttp determines the presence of the `Age` header
 (indicating data that is cached) based on whether the `mtime` is zero;
 however, XrdPfc sets `ctime` to 0 if things aren't cached.

Also set `mtime` to 0 in the not-cached case.
---
 src/XrdPfc/XrdPfc.cc | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/XrdPfc/XrdPfc.cc b/src/XrdPfc/XrdPfc.cc
index b73a97d86..4ab1c5659 100644
--- a/src/XrdPfc/XrdPfc.cc
+++ b/src/XrdPfc/XrdPfc.cc
@@ -1151,8 +1151,9 @@ int Cache::Stat(const char *curl, struct stat &sbuff)
    }
    sbuff.st_size = file_size;
    bool is_cached = DecideIfConsideredCached(file_size, sbuff.st_blocks * 512ll);
-   if ( ! is_cached)
-      sbuff.st_atime = 0;
+   if ( ! is_cached) {
+      sbuff.st_ctime = sbuff.st_mtime = sbuff.st_atime = 0;
+   }
 
    TRACE(Debug, tpfx << "from disk " << curl << " -> " << res);
 
-- 
2.43.5

